cmake/setup_write_config.cmake

```diff
@@ -83,6 +83,12 @@ elseif(DEAL_II_HAVE_CXX17)
   _both("#        C++ language standard:  C++17\n")
 endif()
 
+if(DEAL_II_HAVE_CXX23 OR DEAL_II_HAVE_CXX20)
+  if(DEAL_II_BUILD_CXX20_MODULE)
+    _both("#        Building C++20 module:  ON\n")
+  endif()
+endif()
```

Comment:
What about always printing the value of the CMake variable? It feels weird to only ever to say `ON` and omitting the entry otherwise. Alternatively, just say `Building C++20 module` (without any boolean value following).